### PR TITLE
feat: hint request service and sources

### PR DESCRIPTION
**Pull request by 0xD-Fabio**
## What does this PR change?

This PR implements the underlying mechanism to show Hints during the loading screen.
It contains the HintRequestService, the three types of HintRequestSource, and other collateral classes to manage the Hints retrieval process.

...

## How to test the changes?

Cannot be tested at the moment.

## Our Code Review Standards

https://github.com/decentraland/unity-renderer/blob/master/docs/code-review-standards.md

### <samp>🤖 Generated by Copilot at ba2724e</samp>

This pull request adds the scripts and assets for the new loading screen feature, which displays hints to the user while the scene is loading. The feature uses different sources of hints, such as local files or remote URLs, and selects the best hints based on their priority and availability. The feature also downloads the textures for the hints and displays them on the screen. The pull request creates the `LoadingScreenV2` folder and its subfolders, and adds the interface definitions, the concrete classes, the scriptable object, and the meta files for the loading screen feature.

### CODE REVIEW

File: unity-renderer/Assets/Scripts/MainScripts/DCL/Controllers/LoadingScreenV2/Tests/HintRequestServiceShould.cs [hunk 1]
```diff
@@ -0,0 +1,126 @@
...
+using System.Collections.Generic;
+using System.Linq;
+using System.Threading;
+using System.Threading.Tasks;
+using UnityEngine;
+
+namespace DCL.Controllers.LoadingScreenV2
+{
+    [Category("EditModeCI")]
+    [TestFixture]
```
Comment (popuz) [inline unity-renderer/Assets/Scripts/MainScripts/DCL/Controllers/LoadingScreenV2/Tests/HintRequestServiceShould.cs:14]: no need of this attribute anymore

**Comment by 0xD-Fabio** [comment]
The PR is 99% done, I just need to put it under a feature flag, and I surely need to review the testing units.

File: unity-renderer/Assets/Scripts/MainScripts/DCL/Controllers/LoadingScreenV2/LoadingScreenV2Plugin.cs [hunk 2]
```diff
@@ -0,0 +1,19 @@
...
+using System.Collections.Generic;
+
+public class LoadingScreenV2Plugin : IPlugin
+{
+    private readonly HintRequestService controller;
+
+    public LoadingScreenV2Plugin()
+    {
+        // TODO: this is temporary. The true Loading Screen V2 controller is not implemented yet.
+        // controller = new HintRequestService(new List<IHintRequestSource>(), new SceneController(), new HintTextureRequestHandler());
```
Comment (Kinerius) [inline unity-renderer/Assets/Scripts/MainScripts/DCL/Controllers/LoadingScreenV2/LoadingScreenV2Plugin.cs:12]: If this is behind a feature flag I think its safe to leave this uncommented

**Review by Kinerius — APPROVED** [review]
Looks very good, great job!

**Review by popuz — APPROVED** [review]
Looks good! 
I saw a lot of small cosmetical/code-style related issues, so I suggest that you run our Cleanup Code for your classes to fix formatting and manually fix namings. You probably can do it by using CustomScope and defining your new folders there
image

**Pull request merged**